clients/wallet/src/resubmissions.rs

```diff
@@ -245,48 +322,74 @@ impl StellarWallet {
+	/// returns true if a transaction already exists and WAS submitted successfully.
 	async fn is_transaction_already_submitted(&self, tx: &Transaction) -> bool {
-		// loop through until the 10th page
-		let mut remaining_page = 0;
+		let tx_sequence_num = tx.seq_num;
+		let default_page_size = SequenceNumber::from(DEFAULT_PAGE_SIZE);
 		let own_public_key = self.public_key();
 
-		while let Ok(transaction) = self.get_all_transactions_iter().await {
-			if remaining_page == MAX_LOOK_BACK_PAGES {
-				break
+		// get the iterator
+		let mut iter = match self.get_all_transactions_iter().await {
+			Ok(iter) => iter,
+			Err(e) => {
+				tracing::warn!("is_transaction_already_submitted(): failed to get iterator: {e:?}");
+				return false
+			},
+		};
+
+		// iterate over the transactions
+		while let Some(response) = iter.next().await {
+			// Make sure that we are the sender and not the receiver because otherwise an
+			// attacker could send a transaction to us with the target memo and we'd wrongly
+			// assume that we already submitted this transaction.
+			if !is_source_account_match(&own_public_key, &response) {
+				// move to the next response
+				continue
 			}
 
-			for response in transaction.records {
-				// Make sure that we are the sender and not the receiver because otherwise an
-				// attacker could send a transaction to us with the target memo and we'd wrongly
-				// assume that we already submitted this transaction.
-				match response.source_account() {
-					// no source account was found; move on to the next response
-					Err(_) => continue,
-					// the wallet's public key is not this response's source account;
-					// move on to the next response
-					Ok(source_account) if !source_account.eq(&own_public_key) => continue,
-					_ => {},
-				}
+			let Ok(source_account_sequence)  = response.source_account_sequence() else {
+				tracing::warn!("is_transaction_already_submitted(): cannot extract sequence number of transaction response: {response:?}");
+				// move to the next response
+				continue
+			};
 
-				// Check that the transaction contains the memo that we want to send.
-				if let Some(response_memo) = response.memo_text() {
-					let Memo::MemoText(tx_memo) = &tx.memo else {
-						continue
-					};
+			// check if the sequence number is the same as this response
+			if tx_sequence_num == source_account_sequence {
+				// Check if the transaction contains the memo we want to send
+				return is_memo_match(tx, &response)
+			}
 
-					if are_memos_eq(response_memo, tx_memo.get_vec()) {
-						return true
-					}
+			// if the sequence number is greater than this response,
+			// no other transaction will ever match with it.
+			if tx_sequence_num > source_account_sequence {
+				break
+			}
+
+			// if the sequence number is greater than the last response of the iterator,
+			// then the transaction is possibly in THIS page
+			if tx_sequence_num >= source_account_sequence.saturating_sub(default_page_size) {
```

Comment:
This cannot be reached with the previous if statement. I think we should switch the order of the two `if` statements.

Reply:
i updated the code so it  does not "assume" the last element; I MADE it check the last element.